storage-manager/src/Replicator.cpp

```diff
@@ -154,6 +154,67 @@ int Replicator::newNullObject(const boost::filesystem::path &filename,size_t len
     return err;
 }
 
+ssize_t Replicator::_pwrite(int fd, const void *data, size_t length, off_t offset)
```

Comment:
_pwrite and _write are drop-in replacements for ::pwrite & ::write, which try harder to write the data intact.